buildkite/data_source_buildkite_pipeline.go

```diff
@@ -0,0 +1,77 @@
+package buildkite
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/shurcooL/graphql"
+)
+
+func dataSourcePipeline() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourcePipelineRead,
```

Comment:
`ReadContext` is interesting, because we use `Read` in the existing resources. Seems like v2 of the terraform sdk deprecated `Read` in favour of `ReadContext`. At some point we should update all the resources to `ReadContext`.